app/code/Magento/CatalogGraphQl/Model/AttributesJoiner.php

```diff
@@ -56,26 +60,70 @@ public function join(FieldNode $fieldNode, AbstractCollection $collection): void
      * Get an array of queried fields.
      *
      * @param FieldNode $fieldNode
+     * @param ResolveInfo $resolverInfo
      * @return string[]
      */
-    public function getQueryFields(FieldNode $fieldNode): array
+    public function getQueryFields(FieldNode $fieldNode, ResolveInfo $resolveInfo): array
     {
         if (null === $this->getFieldNodeSelections($fieldNode)) {
             $query = $fieldNode->selectionSet->selections;
             $selectedFields = [];
             /** @var FieldNode $field */
             foreach ($query as $field) {
-                if ($field->kind === 'InlineFragment') {
-                    continue;
+                if ($field->kind === NodeKind::INLINE_FRAGMENT) {
+                    $inlineFragmentFields = $this->addInlineFragmentFields($resolveInfo, $field);
+                    $selectedFields = array_merge($selectedFields, $inlineFragmentFields);
```

Comment:
Can you run all tests. I think the use of array_merge inside a foreach loop will cause a static failure. Is it possible to refactor this to avoid that?

Reply:
I have moved the array_merge from loop